homeassistant/components/recorder/__init__.py

```diff
@@ -520,6 +522,8 @@ def __init__(
         self._commits_without_expire = 0
         self._keepalive_count = 0
         self._old_states: dict[str, States] = {}
+        self._state_attributes_ids: LRU = LRU(2048)
```

Comment:
Please add a comment explaining the tuning of the cache size. Should it be tuned according to the number of states? Can/should we do that automatically?

Reply:
It is based on reasonable educated guess on how frequently attributes are overlapping and how often state changes occur limited by what seemed like a reasonable amount of memory to use.

Reply:
d7c5e58527f6e990fc57813a4df1e1bff2c8b2b9